src/plugins/layout/stateprovider.ts

```diff
@@ -94,30 +112,83 @@ export function createElementLayoutState(element: Element): IElementLayoutState
     // Ignore
   }
 
-  elementState.layout = null;
   if (rect) {
-    let styles = window.getComputedStyle(element);
-
-    elementState.layout = {
+    layout = {
       x: Math.round(rect.left),
       y: Math.round(rect.top),
       width: Math.round(rect.width),
       height: Math.round(rect.height)
     };
+  }
+  return layout;
+}
 
-    // Check if scroll is possible
-    if (styles["overflow-x"] === "auto"
-                              || styles["overflow-x"] === "scroll"
-                              || styles["overflow-x"] === "hidden"
-                              || styles["overflow-y"] === "auto"
-                              || styles["overflow-y"] === "scroll"
-                              || styles["overflow-y"] === "hidden") {
-      elementState.layout.scrollX = Math.round(element.scrollLeft);
-      elementState.layout.scrollY = Math.round(element.scrollTop);
+function getAttributes(element) {
+  let elementAttributes = element.attributes;
+  let tagName = element.tagName;
+  let stateAttributes: IAttributes = {};
+  for (let i = 0; i < elementAttributes.length; i++) {
+    let attr = elementAttributes[i];
+    let attrName = attr.name.toLowerCase();
+
+    // If it's an image and configuration disallows capturing images then skip src attribute
+    if (tagName === "IMG" && !config.showImages && attrName === "src") {
+      continue;
+    }
+
+    // If we are masking text, also mask it from input boxes as well as alt description
+    if (!config.showText && attributeMaskList.indexOf(attrName) >= 0) {
+      stateAttributes[attr.name] = attr.value.replace(/\S/gi, "*");
+    } else {
+      stateAttributes[attr.name] = attr.value;
     }
   }
+  return stateAttributes;
+}
 
-  return elementState;
+function getStyles(element) {
+    let computed = window.getComputedStyle(element);
+    let style = {};
+
+    if (defaultColor.length === 0) {
+      defaultColor = computed[Styles.Color];
```

Comment:
What is default color here and why do we only set it for, what seems to be, only the first element ?

Reply:
Yeah... because "color" property is inherited. Usually you gather default styles by adding a dummy node, attach it to dom, enumerate computed styles, and dettach. In our case we will cause mutations which we don't want. If we monitor color for whatever is the first element, in practice BODY, we just ensure that we not resending that same value again & again.